packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowPagePagination.ts

```diff
@@ -0,0 +1,224 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams, useSearchParams } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { useObjectNamePluralFromSingular } from '@/object-metadata/hooks/useObjectNamePluralFromSingular';
+import { formatFieldMetadataItemsAsFilterDefinitions } from '@/object-metadata/utils/formatFieldMetadataItemsAsFilterDefinitions';
+import { formatFieldMetadataItemsAsSortDefinitions } from '@/object-metadata/utils/formatFieldMetadataItemsAsSortDefinitions';
+import { generateDepthOneRecordGqlFields } from '@/object-record/graphql/utils/generateDepthOneRecordGqlFields';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { turnSortsIntoOrderBy } from '@/object-record/object-sort-dropdown/utils/turnSortsIntoOrderBy';
+import { recordPositionInternalState } from '@/object-record/record-field/states/recordPositionInternalState';
+import { turnObjectDropdownFilterIntoQueryFilter } from '@/object-record/record-filter/utils/turnObjectDropdownFilterIntoQueryFilter';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { usePrefetchedData } from '@/prefetch/hooks/usePrefetchedData';
+import { PrefetchKey } from '@/prefetch/types/PrefetchKey';
+import { View } from '@/views/types/View';
+import { mapViewFiltersToFilters } from '@/views/utils/mapViewFiltersToFilters';
+import { mapViewSortsToSorts } from '@/views/utils/mapViewSortsToSorts';
+import { isDefined } from '~/utils/isDefined';
+
+export const useRecordShowPagePagination = (
+  propsObjectNameSingular: string,
+  propsObjectRecordId: string,
+) => {
+  const {
+    objectNameSingular: paramObjectNameSingular,
+    objectRecordId: paramObjectRecordId,
+  } = useParams();
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const viewIdQueryParam = searchParams.get('view');
+
+  const recordPositionInternal = useRecoilValue(recordPositionInternalState);
+
+  const [currentRecordIndex, setCurrentRecordIndex] = useState(0);
+  const [hasPreviousRecord, setHasPreviousRecord] = useState(false);
+  const [hasNextRecord, setHasNextRecord] = useState(false);
+  const [objectRecords, setObjectRecords] = useState<ObjectRecord[]>([]);
+  const [totalRecords, setTotalRecords] = useState(0);
+  const [isLoadedRecords, setIsLoadedRecords] = useState(false);
+  const [isLoadingPagination, setIsLoadingPagination] = useState(true);
+  const [hasNextPage, setHasNextPage] = useState<boolean | undefined>();
+
+  const [viewName, setViewName] = useState('');
```

Comment:
I think it would have been great to have a shared hook like useAppPageInfo or something like that (not great naming but I don't have a better idea), so that this can be re-used in the commandBar (we'll want to adapt actions in the commandBar based on the selected view and selected object)